include/ofi_hmem.h

```diff
@@ -118,6 +133,11 @@ ssize_t ofi_copy_to_hmem_iov(enum fi_hmem_iface hmem_iface, uint64_t device,
 			     size_t hmem_iov_count, uint64_t hmem_iov_offset,
 			     void *src, size_t size);
 
+int ofi_hmem_get_handle(enum fi_hmem_iface iface, void *dev_buf, void **handle);
+int ofi_hmem_open_handle(enum fi_hmem_iface iface, void *handle,
+			 uint64_t device, void **ipc_ptr);
+int ofi_hmem_close_handle(enum fi_hmem_iface iface, void *ipc_ptr);
```

Comment:
The terms are confusing me.  What's a handle vs dev_buf vs ipc_ptr?

dev_buf seems like it's just a buf.
Is a handle really a void * that's passed to another app?

open_handle() is getting an ipc_ptr, whatever that is
close_handle() closes the ipc_ptr, not a handle

Can we agree that the GPU APIs are goofy?  It looks like a 'handle' is basically some sort of key that can be passed around.  A process can use the key to retrieve the peer's buffer.  Does that map the buffer into the address space?  What does an ipc_ptr do for an app?

Reply:
It's super goofy. This is essentially what the terms are:
dev_buf -> pointer to memory allocated by the process
handle-> an abstract thing that the interface uses to control access across processes (yes, a key). For CUDA this is some 64 byte internal definition. For L0, this is an fd
ipc_ptr-> what a peer uses to reference memory that was not allocated by this peer, mapped into the address space (I assume).

Reply:
We should define a handle as u64, vs void * (which is only 32-bits on a 32-bit system).  I'd prefer to use the term key or hmem_key over handle, though I think it really should be an fd and cuda is broken and other APIs are goofy for trying to abstract away a fundamental concept like this.  Sigh...

The other APIs depend on how this behaves.  If the memory is mapped into the process address space, I would rename open_handle -> map_key, and close_handle -> unmap or free.

Reply:
Agreed on the u64 definition. Good call.

As for the names, while I would tend to agree that renaming makes more sense, but I do think it's confusing to rename get_handle, open_handle, and close_handle when all 3 existing interfaces (CUDA, L0, and rocr) all use get_handle, open_handle, close_handle. I will follow your call though.

What do you mean "the other APIs depend on how this behaves"?

Reply:
By other APIs I meant open_handle and close_handle. We're not opening or closing the handle.  We have the handle.  We're mapping memory and unmapping it.  Their APIs suck, but I agree, let's match it.  ......  Even though I could map these APIs over a shared file or shared memory, and my handle would be an fd, open_handle would be mmap, and close_handle would be free... but I digress.